### ISSUE TITLE
Move tests inside scot package

### ISSUE THREAD

**Issue by cle1109** [scot-dev/scot#80]
People recommend to have the `tests` package as a subpackage inside the actual package. Therefore, I suggest that we move our `tests` package to `scot/tests`.

**Comment by kazemakase** [comment]
We could do that, but what's the benefit, and who are the people? :)

**Comment by cle1109** [comment]
Right now the tests are not part of the scot package, and if we want them to be part of the package we need to move the tests folder into the scot folder. Let me rephrase my first statement: Many projects have their tests right inside the package, including [scikit-learn](https://github.com/scikit-learn/scikit-learn) and [mne](https://github.com/mne-tools/mne-python).

**Comment by mbillingr** [comment]
Ok, but what's the reasoning behind it?
When I install a package I'm usually not interested in the tests that's why I originally put them in an extra directory.

**Comment by cle1109** [comment]
OK, makes sense. I guess we'll leave it as it is then.

**Comment by agramfort** [comment]
>
> Ok, but what's the reasoning behind it?
> When I install a package I'm usually not interested in the tests that's
> why I originally put them in an extra directory.
>

you want your users to be able to run the tests on their local setup
without a "git clone"

**Comment by mbillingr** [comment]
True.. if something is not working we could tell them to run the tests.

**Comment by agramfort** [comment]
Exactly

**Comment by cle1109** [comment]
On the other hand, we could say the same thing about our examples. It would be nice to have them in the package too, but then again, it would also be nice to have the examples and tests neatly separated from the package.

I guess it's up to us to decide - there's no right or wrong.

**Comment by agramfort** [comment]
>
> > there's no right or wrong.
>
indeed it's just a matter of convention between all scientific python
packages.

**Comment by mbillingr** [comment]
I'm starting to see that everything makes sense in the big picture :)

Examples and tests are often put into the docstrings of functions, so in some sense they are included in the functions.
On a larger scale, it seems consequent to include examples and tests in packages too.

**Comment by cle1109** [comment]
Do we have a decision on this one? If not, I suggest not to do anything at the moment (i.e. to keep the tests outside of the package together with the examples).

**Comment by mbillingr** [comment]
I think we should move them into the package in the future. For now, let's focus on getting #65 done.

**Comment by cle1109** [comment]
We should move the tests - most projects are doing this, so we should go with the majority here. Also, this is a nice template that also recommends putting the tests inside the package: https://github.com/uwescience/shablona/
